diff --git a/Makefile.am b/Makefile.am
index 46c7d7cc40..6e6fc70302 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -172,12 +172,6 @@ cppcheck:
 docs:
 	$(MAKE) -C doc/sphinx
 
-# These steps are necessary during installation
-install-exec-hook:
-	mkdir -p $(DESTDIR)${localstatedir}/log/
-	mkdir -p $(DESTDIR)${localstatedir}/lib/${PACKAGE_NAME}
-	mkdir -p $(DESTDIR)${runstatedir}/${PACKAGE_NAME}
-
 add-changelog-entry:
 	@./changelog_unreleased/.add-entry.sh
 
diff --git a/src/lib/dhcpsrv/Makefile.am b/src/lib/dhcpsrv/Makefile.am
index 7e0f3c4..08c53d8 100644
--- a/src/lib/dhcpsrv/Makefile.am
+++ b/src/lib/dhcpsrv/Makefile.am
@@ -420,6 +420,3 @@ libkea_dhcpsrv_parsers_include_HEADERS = \
 	parsers/shared_networks_list_parser.h \
 	parsers/simple_parser4.h \
 	parsers/simple_parser6.h
-
-install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(dhcp_data_dir)
